components/eamxx/src/control/atmosphere_driver.cpp

```diff
@@ -1027,9 +1045,79 @@ void AtmosphereDriver::set_initial_conditions ()
   }
   m_atm_logger->debug("    [EAMxx] Processing subfields ... done!");
 
+  // Load topography from file if topography file is given.
+  if (ic_pl.isParameter("topography_filename")) {
+    m_atm_logger->debug("    [EAMxx] Reading topography from file ...");
+    const auto& file_name = ic_pl.get<std::string>("topography_filename");
+    for (const auto& it : m_field_mgrs) {
+      const auto& grid_name = it.first;
+      read_fields_from_file (topography_fields_names_nc[grid_name],
+                              topography_fields_names_eamxx[grid_name],
+                            it.first,file_name,m_current_ts);
+    }
+    m_atm_logger->debug("    [EAMxx] Processing topography from file ... done!");
+  } else {
+    // Ensure that, if no topography_filename is given, no
+    // processes is asking for topography data (assuming a
+    // separate IC param entry isn't given for the field).
+    for (const auto& it : m_field_mgrs) {
+      const auto& grid_name = it.first;
+      EKAT_REQUIRE_MSG(topography_fields_names_nc[grid_name].size()==0,
+                      "Error! Topography data was requested in the FM, but no "
+                      "topography_filename or entry matching the field name "
+                      "was given in IC parameters.\n");
+    }
+  }
+
   m_atm_logger->info("  [EAMxx] set_initial_conditions ... done!");
 }
 
+void AtmosphereDriver::
+read_fields_from_file (const std::vector<std::string>& field_names_nc,
```

Comment:
There's a lot of code duplication between this fcn and the one below. Can we make the one below call this one, like

```
read_field_from_file (field_names,field_names,grid_name,file_name,t0);
```
?

Reply:
Yeah, let me look at this again. I would also like to add in the stuff from read lat/lon/vcoords from this file as well.